src/Services/Device.vala

```diff
@@ -227,7 +227,30 @@ public class Power.Services.Device : Object {
     }
 
     public string get_symbolic_icon_name_for_battery () {
-        return get_icon_name_for_battery () + "-symbolic";
+        if (!is_a_battery) {
+            return "preferences-system-power-symbolic";
+        }
+
+        var icon_name = "battery";
+
+        if (percentage > 10) {
+            // Round to the nearest 5 percent
+            // Clamp to 20. There is no 15. Make sure we don't have single px red line until < 10
+            var rounded_percentage = (int) (5 * Math.round (percentage / 5)).clamp (20, 100);
+            icon_name += "-%i".printf (rounded_percentage);
+        } else if (percentage > 0) {
+            icon_name += "-10";
+        } else {
+            icon_name += "-0";
```

Comment:
should this be `battery-low` and `battery-caution`? Those are the only battery names in [the spec](https://specifications.freedesktop.org/icon-naming-spec/latest/#names), but they are defined as being < 40% and < 20% respectively, which is obviously pretty outdated in terms of what to expect from modern battery life so I highly doubt anyone is respecting those definitions